packages/meteor/dynamics_browser.js

```diff
@@ -36,11 +36,23 @@ EVp.getOrNullIfOutsideFiber = function () {
  */
 EVp.withValue = function (value, func) {
   var saved = currentValues[this.slot];
+
   try {
     currentValues[this.slot] = value;
+
     var ret = func();
+    var isPromise = Meteor._isPromise(ret);
+
+    if (isPromise) {
+      var self = this;
+      return ret.finally(function () {
```

Comment:
The meteor package could run in environments that have promises, but does not have Promise.finally. The `promise` package would polyfill it, but apps aren't required to use it.